examples/temperaturesensor/temperaturesensor_HTU21D

```diff
@@ -0,0 +1,171 @@
+#endif 
+
+
+#include <Arduino.h>
+#ifdef ESP8266 
+       #include <ESP8266WiFi.h>
+#endif 
+#ifdef ESP32   
+       #include <WiFi.h>
+#endif
+
+#include "SinricPro.h"
+#include "SinricProTemperaturesensor.h"
+#include <Adafruit_HTU21DF.h>
+#include <Wire.h>  
+
+
+Adafruit_HTU21DF htu = Adafruit_HTU21DF();
+
+#define WIFI_SSID         "YOUR-WIFI-SSID"    
+#define WIFI_PASS         "YOUR-WIFI-PASSWORD"
+#define APP_KEY           "YOUR-APP-KEY"      // Should look like "de0bxxxx-1x3x-4x3x-ax2x-5dabxxxxxxxx"
+#define APP_SECRET        "YOUR-APP-SECRET"   // Should look like "5f36xxxx-x3x7-4x3x-xexe-e86724a9xxxx-4c4axxxx-3x3x-x5xe-x9x3-333d65xxxxxx"
+#define TEMP_SENSOR_ID    "YOUR-DEVICE-ID"    // Should look like "5dc1564130xxxxxxxxxxxxxx"
+#define BAUD_RATE         9600                // Change baudrate to your need (used for serial monitor)
+#define EVENT_WAIT_TIME   60000   
+
+#ifdef ESP8266
+      #define I2C_SCL 14    //D5                
+      #define I2C_SDA 12    //D6
+#endif
+#ifdef ESP32
+      #define I2C_SCL 18                    
+      #define I2C_SDA 19 
+#endif
+
+
+bool deviceIsOn;                              // Temeprature sensor on/off state
+float temperature;                            // actual temperature
+float humidity;                               // actual humidity
+float lastTemperature;                        // last known temperature (for compare)
+float lastHumidity;                           // last known humidity (for compare)
+unsigned long lastEvent = (-EVENT_WAIT_TIME); // last time event has been sent
+
+/* bool onPowerState(String deviceId, bool &state) 
+ *
+ * Callback for setPowerState request
+ * parameters
+ *  String deviceId (r)
+ *    contains deviceId (useful if this callback used by multiple devices)
+ *  bool &state (r/w)
+ *    contains the requested state (true:on / false:off)
+ *    must return the new state
+ * 
+ * return
+ *  true if request should be marked as handled correctly / false if not
+ */
+bool onPowerState(const String &deviceId, bool &state) {
+  Serial.printf("Temperaturesensor turned %s (via SinricPro) \r\n", state?"on":"off");
+  deviceIsOn = state; // turn on / off temperature sensor
+  return true; // request handled properly
+}
+
+/* handleTemperatatureSensor()
+ * - Checks if Temperaturesensor is turned on
+ * - Checks if time since last event > EVENT_WAIT_TIME to prevent sending too much events
+ * - Get actual temperature and humidity and check if these values are valid
+ * - Compares actual temperature and humidity to last known temperature and humidity
+ * - Send event to SinricPro Server if temperature or humidity changed
+ */
+void handleTemperaturesensor() {
+  if (deviceIsOn == false) return; // device is off...do nothing
+
+  unsigned long actualMillis = millis();
+  if (actualMillis - lastEvent < EVENT_WAIT_TIME) return; //only check every EVENT_WAIT_TIME milliseconds
+    if (!htu.begin()) 
+          {
+              while (1) {}
+          }
```

Comment:
Hi JPA-Welten and thanks for your contribution!

Lines 101 to 104 would block the whole sketch and other SinricPro functions if the sensor could not be initialized.
Better would be to print an error message here and end the function using `return`.
```suggestion
  if (!htu.begin()) {
    Serial.printf("Sensor not initialized\r\n");
    return;
  }
```